src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/AbstractProvideExecutionFilesStep.groovy

```diff
@@ -0,0 +1,164 @@
+import de.tracetronic.jenkins.plugins.ecutestexecution.configs.PublishConfig
+import de.tracetronic.jenkins.plugins.ecutestexecution.security.ControllerToAgentCallableWithTimeout
+import de.tracetronic.jenkins.plugins.ecutestexecution.util.PathUtil
+import hudson.AbortException
+import hudson.EnvVars
+import hudson.Launcher
+import hudson.model.Result
+import hudson.model.Run
+import hudson.model.TaskListener
+import org.jenkinsci.plugins.workflow.steps.Step
+import org.jenkinsci.plugins.workflow.steps.StepContext
+import org.jenkinsci.plugins.workflow.steps.StepExecution
+import org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution
+import org.kohsuke.stapler.DataBoundSetter
+
+import javax.annotation.Nonnull
+
+abstract class AbstractProvideExecutionFilesStep extends Step implements Serializable {
+    protected String iconName
+    protected String outDirName
+    protected String supportVersion
+    protected PublishConfig publishConfig
+
+    AbstractProvideExecutionFilesStep() {
+        super()
+        this.publishConfig = new PublishConfig()
+    }
+
+    @Nonnull
+    PublishConfig getPublishConfig() {
+        return new PublishConfig(publishConfig)
+    }
+
+    @DataBoundSetter
+    void setPublishConfig(PublishConfig publishConfig) {
+        this.publishConfig = publishConfig ?: new PublishConfig()
+    }
+
+    @Override
+    StepExecution start(StepContext context) throws Exception {
+        return new Execution(this, context)
+    }
+
+    protected abstract ArrayList<String> processReport(File reportZip, String reportDirName, String outDirPath, TaskListener listener)
+
+    static class Execution extends SynchronousNonBlockingStepExecution<Void> {
+        private static final long serialVersionUID = 1L
+
+        private final transient AbstractProvideExecutionFilesStep step
+
+        Execution(AbstractProvideExecutionFilesStep step, StepContext context) {
+            super(context)
+            this.step = step
+        }
+
+        @Override
+        protected Void run() throws Exception {
+            Run run = context.get(Run.class)
+            TaskListener listener = context.get(TaskListener.class)
+            long startTimeMillis = run.getStartTimeInMillis()
+            String outDirPath = PathUtil.makeAbsoluteInPipelineHome(step.outDirName, context)
+
+            try {
+                ArrayList<String> filePaths = context.get(Launcher.class).getChannel().call(
+                        new ExecutionCallable(step.publishConfig.timeout, startTimeMillis, context.get(EnvVars.class), outDirPath, listener, step)
+                )
+                def result = new ProvideFilesBuilder(context).archiveFiles(filePaths, step.outDirName, step.publishConfig.keepAll, step.iconName)
+                if (!result && !step.publishConfig.allowMissing) {
+                    throw new Exception("Missing ${step.outDirName} aren't allowed by step property. Set build result to ${Result.FAILURE.toString()}")
```

Comment:
Build Result set to ${Result.FAILURE.toString()} due to missing ${step.outDirName}. Adjust AllowMissing step property if this is not intended.